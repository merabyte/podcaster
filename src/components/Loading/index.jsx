import PropTypes from 'prop-types'

function Loading({ className }) {
  return <div className={`blob ${className || ''}`}></div>
}

Loading.propTypes = {
  className: PropTypes.string,
}

export default Loading
