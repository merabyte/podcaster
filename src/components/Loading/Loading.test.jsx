import { render } from '@testing-library/react'

import Loading from './'

describe('Loading', () => {
  it('should match snapshot', () => {
    const { container } = render(<Loading />)
    expect(container).toMatchSnapshot()
  })

  it('should render the component', () => {
    const { container } = render(<Loading />)
    const loadingElement = container.querySelector('.blob')

    expect(loadingElement).toBeInTheDocument()
  })
})
