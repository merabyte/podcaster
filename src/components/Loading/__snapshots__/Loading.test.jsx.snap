// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading > should match snapshot 1`] = `
<div>
  <div
    class="lds-ellipsis "
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
